lib/compat/wordpress-6.1/get-global-styles-and-settings.php

```diff
@@ -42,3 +42,77 @@ function ( $item ) {
 		}
 	}
 }
+
+/**
+ * Returns the stylesheet resulting of merging core, theme, and user data.
+ *
+ * @param array $types Types of styles to load. Optional.
+ *                     It accepts 'variables', 'styles', 'presets' as values.
+ *                     If empty, it'll load all for themes with theme.json support
+ *                     and only [ 'variables', 'presets' ] for themes without theme.json support.
+ *
+ * @return string Stylesheet.
+ */
+function gutenberg_get_global_stylesheet( $types = array() ) {
```

Comment:
Ah, apologies, I should have left a comment here. The main change is this:

```
	if ( empty( $types ) && ! $supports_theme_json ) {
		$types = array( 'variables', 'presets' );
		if ( current_theme_supports( 'wp-block-styles' ) ) {
			$types[] = 'base-layout-styles';
		}
```

For Classic themes, we need to explicitly tell the theme JSON class to output some layout styles so that blocks that use layouts have all the required styles to render correctly. I'm not sure if this is the best way to do it, but it was the least awkward way I could work out how to do it 😅 (conditionally ensure that classic themes can get the styles that they need to render core blocks, if they've opted-in to get core blocks' styles).